Library/Homebrew/test/test_language_python.rb

```diff
@@ -71,18 +73,24 @@ def test_pip_install_accepts_resource
     @venv.pip_install res
   end
 
-  def test_link_scripts_links_scripts
-    bin = (@dir/"bin")
-    dest = (@dir/"dest")
-    bin.mkpath
-    refute((bin/"kilroy").exist?)
-    refute((dest/"kilroy").exist?)
+  def test_pip_install_and_link_links_scripts
+    bin = (@dir/"bin").tap(&:mkpath)
+    dest = @formula.dest
+
+    refute_predicate bin/"kilroy", :exist?
+    refute_predicate dest/"kilroy", :exist?
+
     FileUtils.touch bin/"irrelevant"
-    @venv.link_scripts(dest) { FileUtils.touch bin/"kilroy" }
-    assert((bin/"kilroy").exist?)
-    assert((dest/"kilroy").exist?)
-    assert((dest/"kilroy").symlink?)
+    @formula.expects(:system).returns(true).with do |*params|
+      FileUtils.touch bin/"kilroy"
```

Comment:
Can you suggest the correct way how I can make the underlying `pip_install` invocation produce a side effect? Aside from that, I guess in this test I should be doing something like `@venv.expects(:pip_install).yields(nil)` instead …